File: LeaguerManagement.Entities/ViewModels/Leaguers/LeaguerModel.cs
using System;
using System.Collections.Generic;

namespace LeaguerManagement.Entities.ViewModels
{
    public class LeaguerModel
    {

        public LeaguerModel()
        {
            OfficialDocuments = new List<ReferenceWithAttachmentModel<AppliedDocumentModel>>();
        }
        public int Id { get; set; }
        public int UnitId { get; set; }
        public string Name { get; set; }
        public int? AvatarId { get; set; }
        public int BornYear { get; set; }
        public bool Gender { get; set; }
        public string Religion { get; set; }
        public string Folk { get; set; }
        public string HomeTown { get; set; }
        public string EducationalLevel { get; set; }
        public string PoliticalTheoryLevel { get; set; }
        public string ForeignLanguageLevel { get; set; }
        public string SpecializeMajor { get; set; }
        public string BeforeEnteringCareer { get; set; }
        public string CurrentCareer { get; set; }
        public string Position { get; set; }
        public DateTime? PreliminaryApplyDate { get; set; }
        public DateTime? OfficialApplyDate { get; set; }
        public string CardNumber { get; set; }
        public string BackgroundNumber { get; set; }
        public string BadgeNumber { get; set; }
        public DateTime? MoveOutDated { get; set; }
        public string OfficeComing { get; set; }
        public DateTime? MoveInDated { get; set; }
        public string AtOffice { get; set; }
        public DateTime? DeadDate { get; set; }
        public string DeathReason { get; set; }
        public DateTime? GetOutDate { get; set; }
        public string FormOut { get; set; }
        public int? Phone { get; set; }
        public string Notes { get; set; }
        public int StatusId { get; set; }
        public bool IsActivated { get; set; }

        public IList<ReferenceWithAttachmentModel<AppliedDocumentModel>> OfficialDocuments { get; set; }
        public IList<RatingResultModel> RatingResults { get; set; }
    }
}

File: LeaguerManagement.UI/src/app/models/leaguers/leaguer.model.ts
import {ReferenceWithAttachmentModel} from '@app/models/shared/attachment.model';
import {AppliedDocumentModel} from '@app/models/leaguers/applied-document.model';

export class LeaguerModel {
  id: number;
  unitId: number;
  name: string;
  bornYear: number;
  gender: boolean;
  religion: string;
  folk: string;
  homeTown: string;
  educationalLevel: string;
  politicalTheoryLevel: string;
  foreignLanguageLevel: string;
  specializeMajor: string;
  beforeEnteringCareer: string;
  currentCareer: string;
  position: string;
  preliminaryApplyDate: Date | null;
  officialApplyDate: Date | null;
  cardNumber: string;
  backgroundNumber: string;
  badgeNumber: string;
  moveOutDated: string;
  officeComing: string;
  moveInDated: string;
  atOffice: string;
  deadDate: string;
  deathReason: string;
  getOutDate: string;
  formOut: string;
  phone: number;
  notes: string;
  statusId: number;
  isActivated: boolean;
  //
  avatarId: number;
  avatarImg: string;
  //
  officialDocuments: ReferenceWithAttachmentModel<AppliedDocumentModel>[] = [];

  constructor(init?: Partial<LeaguerModel>) {
    Object.assign(this, init);
  }

  public static copy(source: LeaguerModel, destination: LeaguerModel) {
    if (!source || !destination) {
      return destination;
    }

    Object.assign(destination, source);

    return destination;
  }

  public static clone(source: LeaguerModel): LeaguerModel {
    return LeaguerModel.copy(source, new LeaguerModel());
  }
}
